File: 代码随想录/leetcode/贪心算法/455分发饼干.cpp
#include "../header.h"

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        
        int res = 0;
        int i = 0, j = 0;
        while (i < g.size() && j < s.size()) {
            if (s[j] >= g[i]) {
                res++;
                j++;
                i++;
            } else {
                j++;
            }
        }

        return res;
    }
};

File: go/leetcode/455分发饼干.go
func findContentChildren(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)

	res := 0
	i, j := 0, 0
	for i < len(g) && j < len(s) {
		if s[j] >= g[i] {
			res++
			j++
			i++
		} else {
			j++
		}
	}

	return res
}